File: 1510-FindLuckyIntegerInAnArray/1510-FindLuckyIntegerInAnArray.cpp
// Last updated: 7/21/2026, 7:08:54 PM
#include <vector>
#include <unordered_map>
#include <algorithm>
#include <iostream>

class Solution {
public:
    int findLucky(std::vector<int>& arr) {
        std::unordered_map<int, int> counts;

       
        for (int num : arr) { 
            counts[num]++;
        }

        int largestLucky = -1;

        
        for (auto const& [num, count] : counts) { 
            if (num == count) { 
                largestLucky = std::max(largestLucky, num); 
            }
        }

        return largestLucky;
    }
};

File: 1510-FindLuckyIntegerInAnArray/1510-FindLuckyIntegerInAnArray.py
# Last updated: 7/21/2026, 7:08:50 PM
from collections import Counter

class Solution(object):
    def findLucky(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        counts = Counter(arr)
        lucky = -1
        for num, count in counts.items():
            if num == count:
                lucky = max(lucky, num)
        return lucky